sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpAnnotatedMessageExtensions.cs

```diff
@@ -0,0 +1,408 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Azure.Core.Amqp;
+
+namespace Azure.Messaging.EventHubs.Amqp
+{
+    /// <summary>
+    ///   The set of extension methods for the <see cref="AmqpAnnotatedMessage" /> class.
+    /// </summary>
+    ///
+    internal static class AmqpAnnotatedMessageExtensions
+    {
+        /// <summary>
+        ///   Populates the <paramref name="instance"/> from a set of well-known <see cref="EventData"/>
+        ///   attributes, setting only those values which are populated.
+        /// </summary>
+        ///
+        /// <param name="instance">The instance that this method was invoked on.</param>
+        /// <param name="properties">The set of free-form application properties to send with the event.</param>
+        /// <param name="sequenceNumber">The sequence number assigned to the event when it was enqueued in the associated Event Hub partition.</param>
+        /// <param name="offset">The offset of the event when it was received from the associated Event Hub partition.</param>
+        /// <param name="enqueuedTime">The date and time, in UTC, of when the event was enqueued in the Event Hub partition.</param>
+        /// <param name="partitionKey">The partition hashing key applied to the batch that the associated <see cref="EventData"/>, was sent with.</param>
+        /// <param name="lastPartitionSequenceNumber">The sequence number that was last enqueued into the Event Hub partition.</param>
+        /// <param name="lastPartitionOffset">The offset that was last enqueued into the Event Hub partition.</param>
+        /// <param name="lastPartitionEnqueuedTime">The date and time, in UTC, of the event that was last enqueued into the Event Hub partition.</param>
+        /// <param name="lastPartitionPropertiesRetrievalTime">The date and time, in UTC, that the last event information for the Event Hub partition was retrieved from the service.</param>
+        ///
+        public static void PopulateFromEventProperties(this AmqpAnnotatedMessage instance,
```

Comment:
I'm not opposed, but don't we normally use `FromXXX` as a factory method, where the type is created fresh?   This is an extension to an existing instance that populates some values, which seems like it may not fit that naming pattern.    Thoughts?

Reply:
Ah ok, I missed the part about just filling out the existing instance. Agreed that the `FromXXX` doesn't make sense here.